File: calc.c
#include <stdio.h>
#include <ctype.h>

int main() {
    char select;
    float num1, num2, result;

    while (1) {
        printf("Select an operation (+, -, *, /) or 'c' to cancel: ");
        scanf(" %c", &select);

        select = tolower(select);

        if (select == 'c') {
            break;
        }

        printf("Enter first number: ");
        scanf("%f", &num1);
        printf("Enter second number: ");
        scanf("%f", &num2);

        if (select == '+') {
            result = num1 + num2;
        } else if (select == '-') {
            result = num1 - num2;
        } else if (select == '*') {
            result = num1 * num2;
        } else if (select == '/') {
            if (num2 == 0) {
                printf("Cannot divide by zero\n");
                continue;  
            }
            result = num1 / num2;
        } else {
            printf("Invalid operation. Please try again.\n");
            continue;
        }

        printf("Result: %.3f\n", result);
    }

    printf("Done!\n");
    return 0;
}

File: calc.py
def main():
    while True:
        select = input("Select an operation (+, -, *, /) or 'c' to cancel: ").lower()
        if select == 'c':
            break

        num1 = float(input("Enter first number: "))
        num2 = float(input("Enter second number: "))

        if select == '+':
            result = num1 + num2
        elif select == '-':
            result = num1 - num2
        elif select == '*':
            result = num1 * num2
        elif select == '/':
            if num2 == 0:
                print("Cannot divide by zero")
                continue
            result = num1 / num2
        else:
            print("Invalid operation. Please try again.")
            continue

        print(f"Result: {result:.3f}")

if __name__ == "__main__":
    main()
